adapter/internal/oasparser/model/api_operation.go

```diff
@@ -37,34 +44,135 @@ type Operation struct {
+								Value: example,
+							})
+						}
+					}
+				}
+				if len(mockedResponse.Content) > 0 {
+					mockedAPIConfig.Responses = append(mockedAPIConfig.Responses, mockedResponse)
+				}
+			}
+		}
+		if len(mockedAPIConfig.Responses) > 0 {
+			operation.mockedAPIConfig = mockedAPIConfig
+		}
+	}
 }
 
-// MockedHeaderConfig holds header configurations in the mocked API operation
-type MockedHeaderConfig struct {
-	Name  string `json:"name,omitempty"`
-	Value string `json:"value,omitempty"`
+// SetMockedAPIConfigOAS2 generate mock impl endpoint configurations
+func (operation *Operation) SetMockedAPIConfigOAS2(openAPIOperation *spec.Operation) {
+	if openAPIOperation.Responses != nil && len(openAPIOperation.Responses.StatusCodeResponses) > 0 {
+		mockedAPIConfig := &api.MockedApiConfig{
+			Responses: make([]*api.MockedResponseConfig, 0),
+		}
+		// get response codes
+		for responseCode, responseRef := range openAPIOperation.Responses.StatusCodeResponses {
+			mockedResponse := &api.MockedResponseConfig{
+				Code:    strconv.Itoa(responseCode),
+				Content: make([]*api.MockedContentConfig, 0),
+			}
+			for mediaType, content := range responseRef.ResponseProps.Examples {
+				//todo(amali) xml payload gen
+				example, err := asJSON(content)
+				if err == nil {
+					mockedResponse.Content = append(mockedResponse.Content, &api.MockedContentConfig{
+						ContentType: mediaType,
+						Examples:    []*api.MockedContentExample{{Ref: "", Body: example}},
+					})
+				}
+			}
+			// swagger does not support header example/examples
+			if len(mockedResponse.Content) > 0 {
+				mockedAPIConfig.Responses = append(mockedAPIConfig.Responses, mockedResponse)
+			}
+		}
+		// get default response examples
+		if openAPIOperation.Responses.Default != nil && len(openAPIOperation.Responses.Default.Examples) > 0 {
+			mockedResponse := &api.MockedResponseConfig{
+				Code:    "default",
+				Content: make([]*api.MockedContentConfig, 0),
+			}
+			for mediaType, content := range openAPIOperation.Responses.Default.Examples {
+				example, err := asJSON(content)
+				if err == nil {
+					mockedResponse.Content = append(mockedResponse.Content, &api.MockedContentConfig{
+						ContentType: mediaType,
+						Examples:    []*api.MockedContentExample{{Ref: "", Body: example}},
+					})
+				}
+			}
+			// swagger does not support header example/examples
+			if len(mockedResponse.Content) > 0 {
+				mockedAPIConfig.Responses = append(mockedAPIConfig.Responses, mockedResponse)
+			}
+		}
+		if len(mockedAPIConfig.Responses) > 0 {
+			operation.mockedAPIConfig = mockedAPIConfig
+		}
+	}
 }
 
-// MockedContentConfig holds mocked content configurations in the mocked API operation
-type MockedContentConfig struct {
-	ContentType string `json:"contentType,omitempty"`
-	Body        string `json:"body,omitempty"`
+// asJSON parse interface to son string. returns error if a null value has passed
+func asJSON(data interface{}) (string, error) {
```

Comment:
Can we change this method name to `convertJson` ? There's a small typo in the comment.